core/domain/suggestion_services_test.py

```diff
@@ -2084,6 +2094,1217 @@ def test_create_translation_contribution_stats_from_model(self) -> None:
+        """Sets up required topics and stories for translations.
+
+        Returns:
+            Dict[str, str]. A dictionary of the change object for the
+            translations.
+        """
+        explorations = [self.save_new_valid_exploration(
+            '%s' % i,
+            self.owner_id,
+            title='title %d' % i,
+            category=constants.ALL_CATEGORIES[i],
+            end_state_name='End State',
+            correctness_feedback_enabled=True
+        ) for i in range(3)]
+
+        for exp in explorations:
+            self.publish_exploration(self.owner_id, exp.id)
+
+        topic_id = '0'
+        topic = topic_domain.Topic.create_default_topic(
+            topic_id, 'topic_name', 'abbrev', 'description', 'fragm')
+        skill_id_0 = 'skill_id_0'
+        skill_id_1 = 'skill_id_1'
+        self._publish_valid_topic(topic, [skill_id_0, skill_id_1])
+
+        self.create_story_for_translation_opportunity(
+            self.owner_id, self.admin_id, 'story_id_01', topic_id, '0')
+        self.create_story_for_translation_opportunity(
+            self.owner_id, self.admin_id, 'story_id_02', topic_id, '1')
+
+        return {
+            'cmd': 'add_translation',
+            'content_id': 'content',
+            'language_code': 'hi',
+            'content_html': '',
+            'state_name': 'Introduction',
+            'translation_html': '<p>Translation for content.</p>'
+        }
+
+    def test_get_translation_contribution_stats_for_invalid_id_with_strict_true(
+        self
+    ) -> None:
+        with self.assertRaisesRegex(
+            Exception,
+            'The stats models do not exist for the given IDs.'):
+            suggestion_services.get_translation_contribution_stats_models(
+                ['invalid_id'])
+
+    def test_get_translation_review_stats_for_invalid_id_with_strict_true(
+        self
+    ) -> None:
+        with self.assertRaisesRegex(
+            Exception,
+            'The stats models do not exist for the given IDs.'):
+            suggestion_services.get_translation_review_stats_models(
+                ['invalid_id'])
+
+    def test_get_question_contribution_stats_for_invalid_id_with_strict_true(
+        self
+    ) -> None:
+        with self.assertRaisesRegex(
+            Exception,
+            'The stats models do not exist for the given IDs.'):
+            suggestion_services.get_question_contribution_stats_models(
+                ['invalid_id'])
+
+    def test_get_question_review_stats_for_invalid_id_with_strict_true(
+        self
+    ) -> None:
+        with self.assertRaisesRegex(
+            Exception,
+            'The stats models do not exist for the given IDs.'):
+            suggestion_services.get_question_review_stats_models(
+                ['invalid_id'])
+
+    def test_update_translation_contribution_stats_when_submitting(
+        self) -> None:
+        # Steps required in the setup pahse before testing.
```

Comment:
```suggestion
        # Steps required in the setup phase before testing.
```

Make sure to proofread your comments, especially if it will be duplicated. Ditto below.

Reply:
Done